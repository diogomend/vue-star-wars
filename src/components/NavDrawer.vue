<template>
  <v-navigation-drawer v-model="drawer" app>
    <v-list nav>
      <v-list-item two-line>
        <v-list-item-content>
          <v-list-item-title>Diogo Mendon√ßa</v-list-item-title>
          <v-list-item-subtitle>Star Wars app</v-list-item-subtitle>
        </v-list-item-content>
      </v-list-item>

      <v-divider></v-divider>
      <v-list-item-group v-model="group" active-class="text--accent-4">
        <v-list-item
          v-for="(page, index) in pages"
          :key="index"
          :to="page.link"
        >
          <v-list-item-title>{{ page.text }}</v-list-item-title>
        </v-list-item>
      </v-list-item-group>
    </v-list>
  </v-navigation-drawer>
</template>

<script>
export default {
  name: "NavDrawer",
  props: ["show"],
  data() {
    return {
      group: true,
      pages: [
        {
          text: "Home",
          link: "/"
        },
        {
          text: "About",
          link: "/about"
        }
      ]
    };
  },
  computed: {
    drawer: {
      get() {
        return this.show;
      },
      set() {
        this.$emit("toggleDrawer");
      }
    }
  }
};
</script>

<style scoped>
.v-navigation-drawer--absolute {
  z-index: 10;
}
</style>
