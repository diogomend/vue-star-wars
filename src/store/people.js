import swapi from "@/apis/swapi";
import storageHelper from "@/helpers/localStorage";

const actions = {
  fetchPeople: async ({ commit }) => {
    const storedPeople = storageHelper.getFromStorage();

    if (storedPeople && storedPeople.length) {
      commit("SET_PEOPLE", JSON.parse(storedPeople));
      return;
    }
    await getAllPeople(commit);
  }
};

const mutations = {
  SET_PEOPLE(state, people) {
    state.people = people;
  }
};

const getters = {
  getPeople: state => state.people
};

const state = () => ({
  people: []
});

export default {
  namespaced: true,
  getters,
  actions,
  state,
  mutations
};

export const getAllPeople = async commit => {
  let currentPage = 1;
  let hasNextPage = true;
  let people = [];

  while (hasNextPage) {
    const { data } = await swapi.get("people", {
      params: {
        page: currentPage
      },
      useCache: true
    });

    people = people.concat(data.results);
    storageHelper.setItem(JSON.stringify(people));
    commit("SET_PEOPLE", people);

    hasNextPage = !!data.next;
    currentPage += 1;
  }

  return people;
};
