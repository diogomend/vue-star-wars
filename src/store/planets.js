import swapi from "@/apis/swapi";

const actions = {
  showPlanetDetails: async ({ commit }, planetID) => {
    commit("SET_SHOW_PLANET_MODAL", true);
    const { data } = await swapi.get(`planets/${planetID}`);
    commit("SET_SELECTED_PLANET", data);
  },
  hidePlanetModal: async ({ commit }) => {
    commit("SET_SHOW_PLANET_MODAL", false);
  }
};
const mutations = {
  SET_SELECTED_PLANET(state, planet) {
    state.selectedPlanet = planet;
  },
  SET_SHOW_PLANET_MODAL(state, show) {
    state.showPlanetModal = show;
  }
};

const getters = {
  getActivePlanet: state => state.selectedPlanet,
  showPlanetModal: state => state.showPlanetModal
};

const state = () => ({
  selectedPlanet: null,
  showPlanetModal: false
});

export default {
  namespaced: true,
  getters,
  actions,
  state,
  mutations
};
