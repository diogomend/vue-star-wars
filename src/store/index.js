import Vue from "vue";
import Vuex from "vuex";
import people from "@/store/people";
import planets from "@/store/planets";

Vue.use(Vuex);

export const storeOptions = {
  modules: {
    people,
    planets
  }
};

export default new Vuex.Store(storeOptions);
