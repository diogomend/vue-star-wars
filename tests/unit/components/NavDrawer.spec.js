import Vuetify from "@/plugins/vuetify";
import Vue from "vue";
import Vuex from "vuex";
import NavDrawer from "@/components/NavDrawer.vue";
import { mount } from "@vue/test-utils";
import { storeOptions } from "@/store/index";
Vue.use(Vuex);
Vue.use(Vuetify);

describe("NavDrawer.vue", () => {
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
    store = new Vuex.Store(storeOptions);
  });

  it("is a Vue instance", () => {
    const wrapper = mount(NavDrawer, {
      store
    });

    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });

  it("Set drawer should emit event", () => {
    const wrapper = mount(NavDrawer, {
      store
    });

    wrapper.vm.drawer = false;
    expect(Object.keys(wrapper.emitted())[0]).toEqual("toggleDrawer");
  });
});
