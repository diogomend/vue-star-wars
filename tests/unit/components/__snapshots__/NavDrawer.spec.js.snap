// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDrawer.vue is a Vue instance 1`] = `
<v-navigation-drawer app="">
  <v-list nav="">
    <v-list-item two-line="">
      <v-list-item-content>
        <v-list-item-title>Diogo Mendon√ßa</v-list-item-title>
        <v-list-item-subtitle>Star Wars app</v-list-item-subtitle>
      </v-list-item-content>
    </v-list-item>
    <v-divider></v-divider>
    <v-list-item-group active-class="text--accent-4">
      <v-list-item to="/">
        <v-list-item-title>Home</v-list-item-title>
      </v-list-item>
      <v-list-item to="/about">
        <v-list-item-title>About</v-list-item-title>
      </v-list-item>
    </v-list-item-group>
  </v-list>
</v-navigation-drawer>
`;
